src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -57,19 +107,95 @@ internal ConfigurationClientManager(IList<ConfigurationClientWrapper> clients)
             _clients = clients;
         }
 
-        public IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time)
-        {
-            return _clients.Where(client => client.BackoffEndTime <= time).Select(c => c.Client).ToList();
+        internal Logger Logger 
```

Comment:
I would check out this [.NET design guide](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/property?redirectedfrom=MSDN) which talks about set-only properties.

Reply:
Perhaps validate the set method.

Reply:
Updated following the guide